modules/core/exported/packet.go

```diff
@@ -0,0 +1,46 @@
+package exported
+
+// PacketI defines the standard interface for IBC packets
+type PacketI interface {
+	GetSequence() uint64
+	GetTimeoutHeight() Height
+	GetTimeoutTimestamp() uint64
+	GetSourcePort() string
+	GetSourceChannel() string
+	GetDestPort() string
+	GetDestChannel() string
+	GetData() []byte
+	ValidateBasic() error
+}
+
+// Acknowledgement defines the interface used to return
+// acknowledgements in the OnRecvPacket callback.
+type Acknowledgement interface {
+	Success() bool
+	Acknowledgement() []byte
+}
+
+// CallbackPacketData defines the interface used by ADR 8 implementations
+// to obtain callback addresses associated with a specific packet data type.
+// This is an optional interface which indicates support for ADR 8 implementations.
```

Comment:
added https://github.com/cosmos/ibc-go/pull/3287/commits/523f40b017821c13b750c507e7a361afe66328fb, I just used a link to main. I think that's fine since I don't expect our `docs/architecture` layout to change for a very long time